### ISSUE TITLE
Proposal gui: add qr-code to folder id textfield analog to device-id

### ISSUE THREAD

**Issue by Blackstareye** [syncthing/syncthing#9055]
## The Idea
The "This Device" section on the right side of the dashboard has a link / on click feature on the device id field in the "Identification Column".

![grafik](https://github.com/syncthing/syncthing/assets/33430997/03c884ea-61ab-4cab-ad76-3e0715720a38)

This shows the QR code of this device, which is convenient for the setup of mobile devices (for example).

At the moment the setup  for already available folders on android devices is a bit fuzzy if you want a dedicated folder.
You need to create a folder with the folder-id of the remote folder that you want to add / sync with.
Then the folders will match and everything works fine.
The folder-id is a bit difficult to read with the default fonts. You can easily mix 'l' (L) and 'I' (I) there.

This was the origin of the idea of my proposal.  Syncthing has already the ability that the device id can be shown as a qr code 
(And the android-version can import the qr code there).

So it would be great if this feature would not be limited to the device id. 
So a folder match can be easily made on the remote device.

PS: I have several mobile devices which are connected to 3 syncthing nodes. So for me, it would be a great feature, making the setup process  easier. But I can totally see, that this feature can also help users in general setting up a remote folder that they want to sync with.

## Code

I already looked at the code. Unfortunately it is written in angular and I am not familiar with this framework.
But there is already a template  ''gui/default/syncthing/device/idqrModalView.html" and a call in 
gui/default/syncthing/core/syncthingController.js:
```js
        $scope.showDeviceIdentification = function (deviceCfg) {
            $scope.currentDevice = deviceCfg;
            $('#idqr').modal();
        };
```
From my understanding it should be relatively easy adding this feature analog to the already existing model and template.
But this is just a humble guess from my side.

## Draft

![grafik](https://github.com/syncthing/syncthing/assets/33430997/db516a93-23c9-415a-ab8f-d6e6151e5630)

**Comment by tomasz1986** [comment]
This should be easy to implement, however the question is how useful it will be without adding support for the feature in the Android app as well. Also, on a side note, normally you don't really need to input folder IDs manually, as once all devices are connected and you share a folder between them, there is pop-up notification to add the folder with its ID automatically set there.

**Comment by Blackstareye** [comment]
Yes the android app should implement this as well. But that should also not be too difficult.

About your advice: 
Well the issue is, that the popup in the app is not consistent and buggy. Often it is necessary to use the web-ui interface to see the popup at all. And if you use that popup dialog in the webui you have a less convenient and limited ui experience. Something like typing / selecting the destination folder is a pain without an helpful file-dialog. You have to know the absolute path of your destination folder, which is in android...a pain..

So, I think the popup will be always  buggy or  at least not very soon fixed.
So I come up with that workaround. It is perfect if you have multiple devices and want a quick sync setup.

That is the reason I think, this feature can be, regardless of the already implemented popup, really helpful.

I think fixing the popup in the app is a more difficult task, then including the already existing qr scan functionality to the  new folder dialog.

PS: Thanks for the quick answer :)

**Comment by tomasz1986** [comment]
> Yes the android app should implement this as well. But that should also not be too difficult.

This will actually be quite difficult, as the app is currently in maintenance mode, meaning that it receives compatibility and bug fixes only. Unless someone steps up and implements the feature themselves, no new functionality is going to be added in the foreseeable future.

> Well the issue is, that the popup in the app is not consistent and buggy.

Yeah, I was speaking purely about the Web GUI. The native pop-ups in the app have multiple problems right now 🙁.

> That is the reason I think, this feature can be, regardless of the already implemented popup, really helpful.

Is there a way to scan such a QR code and copy its content easily (without having this functionality implemented in the Syncthing app)?

**Comment by Blackstareye** [comment]
> This will actually be quite difficult, as the app is currently in maintenance mode, meaning that it receives compatibility and bug fixes only. Unless someone steps up and implements the feature themselves, no new functionality is going to be added in the foreseeable future.

True, that could be a problem. But as soon as the feature is available on the web-ui side, this feature can be at least be done on the android part.

> That is the reason I think, this feature can be, regardless of the already implemented popup, really helpful.
Yes as far as I know, there a quite a few apps that can scan qr codes. And afaik it is possible to copy/paste it from there. I think Inkscape use that for something, but not sure. I will look up this.

Regarding the app, well maybe I could do that when I look into the android realm again :D 
it's been a long time since I've been involved with Android but maybe I could find time for that.
But I can't gurantee it. 
But as I already said, as soon as the feature is implemented, it can be done at least on the android side :) 

As a workaround we could use other QR-Apps in the mean time.

**Comment by calmh** [comment]
> The folder-id is a bit difficult to read with the default fonts. You can easily mix 'l' (L) and 'I' (I) there.

This is a side comment, but our generated folder IDs do not contain 0, 1 (one), 8, l (ell) for roughly this reason. Though the parsing isn't as lenient as it is for device IDs (it can't be).
